appengine/pond_storage/get-ducks.py

```diff
@@ -15,14 +15,25 @@
 limitations under the License.
 """
 
-"""Get all ducks for the current user 
+"""Given a duck id get the duck. If no duck id exists get all ducks belonging to the user.
 """
 
 __author__ = "aschmiedt@google.com (Abby Schmiedt)"
 
+import cgi
 import json
 from pond_storage import *
 
-duckList = get_user_ducks()
-print("Content-Type: application/json\n")
-print(json.dumps({"duckList": duckList}))
+forms = cgi.FieldStorage()
+
+if forms.has_key("duckId"):
```

Comment:
I think its confusing to use "duckId" when the value is not the duck id, but the urlSafe version of the key. We also pass the key in the other requests and we use "key" for the same type, so I think this should be "key" instead.

Reply:
I'm not a huge fan of just key since it doesn't describe what it is a key for. What about duckKey?

Reply:
I like the shortness of "key", but I see your point with the descriptiveness. Another idea could be to update the urls so that when we make a request it looks like:
pond-storage/update-duck?key=1234

However, since we are using "key" in all the other files, I'd like to keep it consistent here and address it in a follow up PR to update it everywhere.

Reply:
done